cpp/src/arrow/record_batch_test.cc

```diff
@@ -410,4 +410,32 @@ TEST_F(TestRecordBatchReader, BeginEndForLoop) {
   ASSERT_EQ(i, static_cast<int64_t>(batches_.size()));
 }
 
+ARROW_SUPPRESS_DEPRECATION_WARNING
+TEST_F(TestRecordBatchReader, DeprecatedReadAllToRecordBatches) {
```

Comment:
Should we also test the new (non-deprecated) methods?

Reply:
Probably yes.